src/js/components/Popup.js

```diff
@@ -13,30 +15,44 @@ const Popup = forwardRef((props, ref) => {
     }
   }
 
-  const serializedProps = serializeProps([
-    "loading",
-    "service",
-    "subdomain",
-    "projects",
-    "activities",
-    "schedules",
-    "timedActivity",
-    "serviceLastProjectId",
-    "userLastProjectId",
-    "serviceLastTaskId",
-    "userLastTaskId",
-    "fromDate",
-    "toDate",
-    "errorType",
-    "errorMessage",
-  ])(props)
+  const sendData = useCallback(() => {
+    if (iFrameRef.current) {
+      iFrameRef.current.contentWindow.postMessage(
+        {
+          type: "moco-bx-popup-data",
+          data: JSON.stringify(props.data),
+        },
+        browser.runtime.getURL("popup.html"),
+      )
+    }
+  }, [props, iFrameRef])
+
+  const handlePopupReady = useCallback(
```

Comment:
Needs a `useCallback` in order to have a stable function we can unregister when the component unmounts, see L47 in this file.